src/main/kotlin/org/dif/diddoc/DIDDocResolver.kt

```diff
@@ -0,0 +1,16 @@
+package org.dif.diddoc
+
+import java.util.Optional
+
+/**
+ * Represents DID Doc resolver (https://www.w3.org/TR/did-core/#did-resolution).
+ */
+interface DIDDocResolver {
+    /**
+     * Resolves a DID document by the given DID.
+     * @param did a DID to be resolved.
+     *
+     * @return An instance of resolved DID DOC or null if DID is not found.
+     */
+    fun resolve(did: String): Optional<DIDDoc>
```

Comment:
Here we assume completely synchronous API, right? Did we consider using `CompletableFuture` as a result?

Reply:
I suggest implementing synchronous API at first. then think about async api

Reply:
As an option it's create DIDCommAsync class and separate async and sync API